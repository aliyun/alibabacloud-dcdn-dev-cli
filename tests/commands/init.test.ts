// init.test.js
import { it, describe, expect, vi } from 'vitest';
import { handleInit } from '../../src/commands/init/index.js';
import * as selectInput from '../../src/components/selectInput.js';
import * as descriptionInput from '../../src/components/descriptionInput.js';
import { mockConsoleMethods } from '../helper/mockConsole.js';
import fs from 'fs';

vi.mock('child_process');
vi.mock('fs/promises', () => ({
  rename: vi.fn()
}));

describe('handleInit', () => {
  let std = mockConsoleMethods();
  afterEach(() => {
    vi.clearAllMocks();
  });

  it('clones the repository and updates project config --install git', async () => {
    vi.spyOn(descriptionInput, 'descriptionInput').mockResolvedValue('test');

    vi.spyOn(selectInput, 'default').mockImplementationOnce(
      ({ items, handleSelect }) => {
        handleSelect(items[0]);
        return () => {};
      }
    );

    vi.spyOn(selectInput, 'default').mockImplementationOnce(
      ({ items, handleSelect }) => {
        handleSelect(items[0]);
        return () => {};
      }
    );

    vi.spyOn(fs, 'readdirSync').mockReturnValue(['test' as any]);

    await handleInit({
      _: [],
      $0: ''
    });
    expect(std.out).matchSnapshot();
  });

  it('clones the repository and updates project config -- uninstall git', async () => {
    vi.spyOn(descriptionInput, 'descriptionInput').mockResolvedValue('test');

    vi.spyOn(selectInput, 'default').mockImplementationOnce(
      ({ items, handleSelect }) => {
        handleSelect(items[1]);
        return () => {};
      }
    );

    vi.spyOn(selectInput, 'default').mockImplementationOnce(
      ({ items, handleSelect }) => {
        handleSelect(items[0]);
        return () => {};
      }
    );

    await handleInit({
      _: [],
      $0: ''
    });
    expect(std.out).matchSnapshot();
  });
});
