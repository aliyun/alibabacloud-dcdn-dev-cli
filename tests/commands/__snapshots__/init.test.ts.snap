// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`handleInit > clones the repository and updates project config -- uninstall git 1`] = `
[MockFunction log] {
  "calls": [
    [
      "Repository cloned successfully.",
    ],
    [
      "ðŸ’¬ Do you want to init git in your project?",
    ],
    [
      "ðŸ’¬ Git installation was skipped.",
    ],
    [
      "ðŸ’¬ Do you want to deploy your project?",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`handleInit > clones the repository and updates project config --install git 1`] = `
[MockFunction log] {
  "calls": [
    [
      "Repository cloned successfully.",
    ],
    [
      "ðŸ’¬ Do you want to init git in your project?",
    ],
    [
      "Git has been installed successfully.",
    ],
    [
      "ðŸ’¬ Do you want to deploy your project?",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
